app/models/temp_file.rb

```diff
@@ -3,4 +3,12 @@ class TempFile < ActiveRecord::Base
 
   has_attached_file :file
   do_not_validate_attachment_file_type :file
+
+  def destroy_obsolete_files
+    file = TempFile.find_by_id(self.id)
+    file.destroy! if file
```

Comment:
Can't this be simply
```ruby
def destroy_obsolote
  destroy!
end
handle_asynchronously :destroy_obsolete,
                      run_at: proc { 7.days.from_now }
```
Or you need your queries because it's executed in delayed job?